db-commons/src/main/scala/org/bitcoins/db/AppConfig.scala

```diff
@@ -3,6 +3,11 @@ package org.bitcoins.db
 import org.bitcoins.core.config.NetworkParameters
 import org.bitcoins.core.protocol.blockchain.ChainParams
 
-case class AppConfig(dbConfig: DbConfig, chain: ChainParams) {
-  val network: NetworkParameters = chain.network
-}
+
+trait AppConfig[C <: DbConfig] {
+  def dbConfig: C
+
+  def chain: ChainParams = dbConfig.networkDb.chain
+
+  def network: NetworkParameters = chain.network
+}
```

Comment:
I think it would be super nice to have an `EnvParams` and then construct all `AppConfig`s by taking the `EnvParams` and a `ChainParams` rather than forcing the user to construct a `dbConfig`

Reply:
I think this is out of scope of this PR, we can do this later via a config file.